HHS/Head-Start-TTADP
# [TTAHUB-3078] SSDI - Backend only

## Description of change

This pull request introduces multiple updates and new features:

1. Generic Class for Permissions:
  - A new Generic class to manage user permissions and region access.
  - Methods include canAccessRegion, filterRegions, and getAllAccessibleRegions.

2. SQL Query Updates:
  - Updates to class-goal-dataset.sql, class-goal-use.sql, and communication-logs.sql to handle SSDI flags as arrays, improving data filtering and accuracy.

3. New Route Addition:
  - Adds a new route /api/admin/ssdi with the following API methods:
    - GET /api/admin/ssdi/list-queries: Lists available SSDI queries.
    - GET /api/admin/ssdi/get-flags: Retrieves SSDI flags.
    - GET /api/admin/ssdi/run-query: Runs a specified SSDI query.

## How to test

- Listing Queries (GET /api/admin/ssdi/list-queries):
  - Send a GET request to /api/admin/ssdi/list-queries and verify that the response includes the list of available SSDI queries.

     `curl -X GET "http://localhost:3000/api/admin/ssdi/list-queries"`

- Retrieving Flags (GET /api/admin/ssdi/get-flags):
  - Send a GET request to /api/admin/ssdi/get-flags and verify that the response includes the SSDI flags.
  
    `curl -X GET "http://localhost:3000/api/admin/ssdi/get-flags?path=monthly-delivery-report.sql"`

- Running a Query (GET /api/admin/ssdi/run-query):
  - Send a GET request to /api/admin/ssdi/run-query with the required query parameters. Verify that the query runs correctly and the response includes the expected data.
    
    `curl -X GET "http://localhost:3000/api/admin/ssdi/run-query?path=monthly-delivery-report.sql&format=csv&regionIds[]=1&startDate[]=2024-05-01&startDate[]=2024-05-31"`

    `curl -X GET "http://localhost:3000/api/admin/ssdi/run-query?path=monthly-delivery-report.sql&format=json&regionIds[]=1&startDate[]=2024-05-01&startDate[]=2024-05-31"`

    `curl -X GET "http://localhost:3000/api/admin/ssdi/run-query?path=monthly-delivery-report.sql&regionIds[]=1&startDate[]=2024-05-01&startDate[]=2024-05-31"`

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-0


## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
